src/server/array_func.c

```diff
@@ -569,23 +569,20 @@ get_subjob_state(job *parent, int iindx)
 void
 update_subjob_state_ct(job *pjob)
 {
-	char *buf;
+	char buf[BUF_SIZE];
 	static char *statename[] = {
 		"Transit", "Queued", "Held", "Waiting", "Running",
-		"Exiting", "Expired", "Beginning", "Moved", "Finished" };
-
+		"Exiting", "Expired", "Beginning", "Moved", "Finished"};
 
-	buf = malloc(150);
-	if (buf == NULL)
-		return;
 	buf[0] = '\0';
-	sprintf(buf+strlen(buf), "%s:%d ", statename[JOB_STATE_QUEUED],
-		pjob->ji_ajtrk->tkm_subjsct[JOB_STATE_QUEUED]);
-	sprintf(buf+strlen(buf), "%s:%d ", statename[JOB_STATE_RUNNING],
-		pjob->ji_ajtrk->tkm_subjsct[JOB_STATE_RUNNING]);
-	sprintf(buf+strlen(buf), "%s:%d ", statename[JOB_STATE_EXITING],
-		pjob->ji_ajtrk->tkm_subjsct[JOB_STATE_EXITING]);
-	sprintf(buf+strlen(buf), "%s:%d ", statename[JOB_STATE_EXPIRED],
+	sprintf(buf, "%s:%d %s:%d %s:%d %s:%d",
```

Comment:
Consider turning this into an snprintf().  It is unlikely to be necessary because BUF_SIZE is 512, and that is definitely large enough to store this mall string.